### ISSUE TITLE
Error: Timeout, Rancher is not ready: <nil>

### ISSUE THREAD

**Issue by ahmedfourti** [rancher/terraform-provider-rancher2#305]
Hello,

It seems the new version `1.8.2` doesn't work as expected.
Here is my code for provider

provider "rancher2" {
  alias = "bootstrap"

  api_url   = "https://${var.rancher_hostname}"
  bootstrap = true
}

The variable `rancher_hostname` is built from an outputs of infrastructure.

However, it hangs for some time and then it throws error `Error: Timeout, Rancher is not ready: <nil>`
I tried the retries option as explained here  #292  but still the same issue.

With version `1.8.1` it works fine.

**Comment by mcmcghee** [comment]
Same issue here.

**Comment by rawmind0** [comment]
Got it. Investigating the issue.

**Comment by rawmind0** [comment]
Hi all, could you please provide the tf file your are using??

**Comment by ahmedfourti** [comment]
Here is mine

```
provider "rancher2" {
  alias = "bootstrap"

  api_url   = "https://${var.rancher_hostname}"
  bootstrap = true
}

resource "rancher2_bootstrap" "admin" {
  provider = rancher2.bootstrap

  current_password = var.rancher_current_password
  password         = var.rancher_password
}
```

**Comment by rawmind0** [comment]
@ahmedfourti , is your terraform apply stuck before plan or it's failing when applying?? If it's failing on applying plan, what is the `retries` value you have tried??

**Comment by ahmedfourti** [comment]
@rawmind0  it fails when it tries to refresh, so no matter plan or apply, it fails.
I tried the default retries to 5, then 50 then 100.

**Comment by rawmind0** [comment]
I'm asking you because matters if plan or apply. If stuck before plan, retries doesn't matter, but if fails applying it's a question of retries. It's a problem how terraform init provdiers.

Anyway, what is the initial value of `${var.rancher_hostname}`?? Is it empty and would be generated during same apply??

**Comment by ahmedfourti** [comment]
It is stuck during refresh, so before both plan and apply.
Yes the variable is empty and generated later during the apply

**Comment by rawmind0** [comment]
PR #306 will address this issue. In the meanwhile we test and release, please continue using provider v1.8.1

**Comment by rawmind0** [comment]
PR merged. Fix will be included on v1.8.3. Please reopen issue if needed
